### DISCUSSION TITLE
SimConnect Crashing MSFS

### DISCUSSION THREAD

**Discussion by Andywmm9** [nguyenquyhy/Flight-Tracker-StreamDeck#134] [General]
I know this isn't caused by this program, but from what I gather SimConnect can crash the sim if several apps are using it.  I started having random CTD of MSFS with no error messages, and the timing seemed to be after I started using this.  Has anyone else experienced this?

**Comment by rmroc451** [comment]
I hope that isn't the case, but seems plausible.

I had 1 CTD since launch, and that was after I had this, simair and had just loaded up flight-events.

I don't normally run with flight events, so maybe that was why I hadn't seen more CTDs?

I haven't had a CTD in hours now since turning off live traffic. But it would be interesting to see if I can get the same CTD free time span without running any of these apps at the same time.

**Comment by Andywmm9** [comment]
I never had a CTD before I started using Streamdeck too.  I am running 3 simconnect apps, the real sim gear app for my G1000s, an app I wrote to relay data to foreflight (see my profile), and this app.

**Comment by Crunchmeister1** [comment]
The commonality that CTDs seem to have in the last 2 weeks is the sim update. Simconnect may be what's failing, but it's the sim that's at fault. Asobo changed something and it's causing instability. 

Like many others, I had no CTDs to speak of since launch. But since the last update, I get multiple per day. Most seem to happen on approach, but have happened in cruise, just sitting on the apron with nothing running, etc.

**Reply by Andywmm9** [reply]
Just had one in cruise now at FL190.  I wasn't doing anything at the time, I heard the audio stutter, and then I found myself back at the desktop.  You're right, I never had crashes before this last sim update.

**Reply by rmroc451** [reply]
Same

**Reply by Andywmm9** [reply]
Well, I guess it's back to X-Plane for a while for me.  But after the weather and amazing graphics, it's hard to go back.

**Reply by Andywmm9** [reply]
I've been looking for some sort of log file, does one exist for MSFS?

**Reply by nguyenquyhy** [reply]
I doubt it is SimConnect. Have you tried to empty your Community folder? I always run the sim with Flight Events, Flight Tracker Stream Deck plugin, Navigraph SimLink, sometimes LNM or FSUIPC. All those use SimConnect and I have not get any issues.

Stuff using SimConnect are generally "safe", those in Community folder are not.

**Reply by rmroc451** [reply]
I've scoured event logs in the event viewer. Very sparse

**Reply by Andywmm9** [reply]
I use navigraph, and the working title G1000 mod v0.4 is all.  It's possible the G1000 could be doing it I guess, I'll have to try taking that out.  I wish there were some log files I could look at.

**Reply by rmroc451** [reply]
I don't have anything besides mobi flight in my community folder @nguyenquyhy.

I'm not saying it's simconnect either, just interesting seeing what everyone else has running.

As crunch says, it has to be the last update. But there hasn't been a peep about it from anyone on the forums or other official channels

**Reply by Andywmm9** [reply]
I see this in event viewer, of course it tells us nothing:

`Faulting application name: FlightSimulator.exe, version: 0.0.0.0, time stamp: 0x5fda32ba
Faulting module name: FlightSimulator.exe, version: 0.0.0.0, time stamp: 0x5fda32ba
Exception code: 0xc0000005
Fault offset: 0x000000000119ef92
Faulting process id: 0x2804
Faulting application start time: 0x01d6e2f5ef3fd802
Faulting application path: C:\Program Files\WindowsApps\Microsoft.FlightSimulator_1.12.13.0_x64__8wekyb3d8bbwe\FlightSimulator.exe
Faulting module path: C:\Program Files\WindowsApps\Microsoft.FlightSimulator_1.12.13.0_x64__8wekyb3d8bbwe\FlightSimulator.exe
Report Id: 3ad2c27a-67aa-4504-bc14-7d65e4d63175
Faulting package full name: Microsoft.FlightSimulator_1.12.13.0_x64__8wekyb3d8bbwe
Faulting package-relative application ID: App`

**Reply by Crunchmeister1** [reply]
I've tried playing with no mods, liveries, SD, LNM, etc. Even removed my Navigraph data and didn't use Navigraph at all. It was a completely vanilla experience. And I still got CTDs. 

It's not mods or any add ons. They broke something inn the last update.

**Reply by Crunchmeister1** [reply]
And one of the crashes I had in full vanilla mode was at FL300 in cruise with the TBM. I wasn't even touching the controls and hadn't in some time. It wasn't a user input event that caused the crash. 

They all happen just as Andy described. Sound stutters, screen freezes, and you're staring at the desktop. The crash log is pretty much identical to Andy's with only file paths being different. It means absolutely nothing to anyone outside of Asobo. And they probably have no idea what it means either.

I reported it to Zendesk along with the crash log on laugh day of the patch. My ticket hasn't even been looked at yet.

**Reply by Crunchmeister1** [reply]
So this is odd.

 Since the update, I was seeing a CTD every 3 to 4 flights. I was only running short flights because of that, but it would often crash on approach regardless. 

Now I haven't had a single crash in 2 days. And I've flown a lot, including some longer flights in the TBM. Nothing has changed here. No new mods, hardware, system or even driver updates. I haven't done a thing, yet my crashes are gone. 

Perhaps one of the online services was causing an issue by feeding out invalid data and they fixed it?

**Reply by nguyenquyhy** [reply]
Yeah, or some weird weather condition. That's why Asobo keeps saying that it is very hard for them to be able to test all different conditions on their own.

**Comment by Adi-18** [comment]
Hi. I also have CTD but it can also reproduce with other tools like "Little Navmap" or FSPanel with FSUPC7 (Tool for accessing android panel). Its reported to zendesk.com. I think its a problem of too many requests.

Anser:
Thanks for contacting Microsoft Flight Simulator Support today. There is a know issue since the last patch where FSUPC7 and Little Navmap are causing crashes with the sim. The issue is currently under research.

**Comment by Andywmm9** [comment]
I built myself a new powerful PC last week, and I haven't had one CTD since.  I pulled the old GeForce 1070 out of my old PC and put it in this build, until I can get my hands on a 3080.  From what it seems, if the main thread is bottlenecked too far, CTD are probable.  I'm using the same exact programs/configuration as I did on my last PC.

I still do get times where MSFS won't start every couple of days.  It's always the same file in my AppSettings folder that I have to delete and it fixes the problem.  It's binary data so it's nothing I can look at.  Seems that they need better error handling and a way to reset your cloud settings.  Of course, after I delete this file I then have to reprogram all my settings.  At least joystick configs are in another file.

**Reply by Andywmm9** [reply]
Well, I take that back.  I had 3 CTD last night with my new machine, with no error messages.  All after midway through the flight.  The only thing I changed is my graphics settings from Medium to High.  I'm going to attribute this to the CTD.  It seems that if too much of a load is placed on the sim, it crashes.   It's just quite a shame there are no logs, and no information available in event viewer.  I'm going to go back to medium and verify that the crashing stopped.
